### PR TITLE
SNOW-872482: Fix getPeerCertificate

### PR DESCRIPTION
**Pull request by sfc-gh-pbulawa**
### Description
Fix getPeerCertificate usage.

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [x] Extend the README / documentation and ensure is properly displayed (if necessary)
- [x] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: lib/agent/socket_util.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ exports.secureSocket = function (socket, host, agent, mock)
   const validate = function ()
   {
     // stop listening for the secure event
-    socket.removeListener('secure', validate);
+    socket.removeListener('secureConnect', validate);
```
Comment (sfc-gh-dprzybysz) [inline lib/agent/socket_util.js:75]: let's create a const for this event name

File: lib/agent/socket_util.js [hunk 2]
```diff
@@ -86,10 +86,11 @@ exports.secureSocket = function (socket, host, agent, mock)
     {
       if (!socket.authorized)
       {
-        return socket;
+        Logger.getInstance().error('Socket is not authorized: %s', socket.authorizationError);
```
Comment (sfc-gh-dprzybysz) [inline lib/agent/socket_util.js:89]: we are renewing the socket then? if so the it should be warn, if not close this comment

**Review by sfc-gh-dprzybysz — APPROVED** [review]

File: lib/agent/socket_util.js (hunk 1 above)
Comment (sfc-gh-pbulawa) [reply lib/agent/socket_util.js]: Done.

File: lib/agent/socket_util.js (hunk 2 above)
Comment (sfc-gh-pbulawa) [reply lib/agent/socket_util.js]: Changed to warn.

**Pull request merged**